dist/extras/amd.js

```diff
@@ -3,132 +3,178 @@
+  /**
+   * Define polyfill
+   */
+  global.define = function (name, deps, executor) {
+    // define('', [], function () {})
+    if (typeof name === 'string') {
+      if (amdDefineDependencies) {
+        if (!System.registerRegistry) {
+          throw Error('Include the named register extension for SystemJS named AMD support.');
+        }
+        System.registerRegistry[name] = createAMDRegister(deps, executor);
+        amdDefineDependencies = [];
+        amdDefineExecutor = emptyFunction;       
+        return;
+      } else {
+        if (System.registerRegistry) {
+          System.registerRegistry[name] = createAMDRegister([].concat(deps), executor);
+        }
+        name = deps;
+        deps = executor;
+      }
+    }
+    // define([], function () {})
+    if (name instanceof Array) {
+      amdDefineDependencies = name;
+      amdDefineExecutor = deps;
+    }
+    // define({})
+    else if (typeof name === 'object') {
+      amdDefineDependencies = [];
+      amdDefineExecutor = function () { return name };
+    }
+    // define(function () {})
+    else if (typeof name === 'function') {
+      amdDefineDependencies = requireExportsModule;
+      amdDefineExecutor = name;
+    }
+  };
 
-  function createAMDRegister (amdDefineDeps, amdDefineExec) {
+  /**
+   * Dynamically create an Registry for our AMD-Module
+   */
+  function createAMDRegister (deps, executor) {
     const exports = {};
     const module = { exports: exports };
+
+    const _deps = handleDependencies(deps);
+
+    const dependencies = _deps[0];
+    const depModules = _deps[1];
+    const setters = _deps[2];
+    
+    const _executor = executor;
+    return [dependencies, function (_export) {
+      _export({ default: exports, __useDefault: true });
+      return {
+        setters: setters,
+        execute: function () {
+          module.exports = _executor.apply(exports, depModules) || module.exports;
+          if (exports !== module.exports) {
+            _export('default', module.exports);
+          }
+        }
+      };
+    }];
+  }
+
+  /**
+   * Handle Dependencies of said AMD Registry
+   * @param {*} dependencies 
+   * @returns [dependencies, depModules, setters]
+   */
+  function handleDependencies(dependencies) {
     const depModules = [];
     const setters = [];
     let splice = 0;
-    for (let i = 0; i < amdDefineDeps.length; i++) {
-      const id = amdDefineDeps[i];
+
+    for (let i = 0; i < dependencies.length; i++) {
+      const id = dependencies[i];
       const index = setters.length;
+
       if (id === 'require') {
         depModules[i] = unsupportedRequire;
         splice++;
-      }
-      else if (id === 'module') {
+      } else if (id === 'module') {
         depModules[i] = module;
         splice++;
-      }
-      else if (id === 'exports') {
+      } else if (id === 'exports') {
         depModules[i] = exports;
         splice++;
-      }
-      else {
-        // needed for ie11 lack of iteration scope
```

Comment:
this comment here really confused the hell out of me because it will always be triggered, both on chrome and firefox